pkg/admission/types.go

```diff
@@ -31,3 +32,9 @@ type WantsInformerFactory interface {
 	SetInformerFactory(informers.SharedInformerFactory)
 	Validator
 }
+
+// WantsAuthorizer defines a function which sets Authorizer for admission plugins that need it.
+type WantsAuthorizer interface {
+	SetAuthorizer(authorizer.Authorizer)
+	Validator
```

Comment:
> From what you've linked, it seems like these setters should return errors, PluginInitializer.Initialize should propagate those errors, and then you've eliminated the need for a separate validation step.

As we add more of these, the interactions between them will become significant.  For instance, you can have `(A && B) || (-A && -B)`, but you can't have `-A && B`